src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T item;
+        private Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = (Node<T>) new Node<>(null, value, null);
+            tail = head;
+        }
+        if (size > 0) {
+            Node<T> tempNode = tail;
+            tail = (Node<T>) new Node<>(tempNode, value, null);
+            tempNode.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Invalid Index");
+        }
+        if (size == 0) {
+            head = (Node<T>) new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return;
+        }
+        Node<T> tempNode = (Node<T>) new Node<>(null, value, null);
+
+        if (index == 0) {
+            tempNode.next = head;
+            head.prev = tempNode;
+            head = tempNode;
+            size++;
+            return;
+        }
+        if (index == size) {
+            tempNode.prev = tail;
+            tail.next = tempNode;
+            tail = tempNode;
+            size++;
+            return;
+        }
+        Node<T> prevNode = head;
+        Node<T> nextNode = head;
+
+        for (int i = 0; i < index; i++) {
+            nextNode = nextNode.next;
+            if (i != index - 1) {
+                prevNode = prevNode.next;
+            }
+        }
+        tempNode.prev = prevNode;
+        prevNode.next = tempNode;
+        tempNode.next = nextNode;
+        nextNode.prev = tempNode;
+
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            Node<T> tempNode = tail;
+            tail = (Node<T>) new Node<>(tempNode, list.get(i), null);
+            tempNode.next = tail;
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (size > index && index >= 0) {
+            Node<T> tempNode = head;
+            for (int i = 0; i < index; i++) {
+                tempNode = tempNode.next;
+            }
+            return tempNode.item;
+        } else {
+            throw new IndexOutOfBoundsException("Invalid Index");
+        }
     }
 
     @Override
     public T set(T value, int index) {
```

Comment:
too complex logic, this is how this method logic should look like:
1. Find node by index.
2. Create a variable that will store the current value of found node.
3. Update the value of the current node with a new one.
4. Return old value of a node